### PR TITLE
[Jest] Migrate `source_cache.test.js`

### PR DESCRIPTION
**Pull request by wipfli**
- Move and rename
- Uncomment line that makes jscodeshift fail
- Run jscodeshift
- Run custom codemods and lint
- Manual edits
- Wrap in describe

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: src/source/source_cache.test.ts [hunk 1]
```diff
@@ -54,20 +59,24 @@ function MockSourceType(id, sourceOptions, _dispatcher, eventedParent) {
...
 
-function createSourceCache(options, used) {
+function createSourceCache(options?, used?) {
     const sc = new SourceCache('id', extend({
         tileSize: 512,
         minzoom: 0,
         maxzoom: 14,
         type: 'mock-source-type'
-    }, options), /* dispatcher */ {});
+    }, options), /* dispatcher */ {} as Dispatcher);
```
Comment (HarelM) [inline src/source/source_cache.test.ts:70]: Remove comment of type.

File: src/source/source_cache.test.ts [hunk 2]
```diff
@@ -1232,35 +1202,31 @@ describe('SourceCache#clearTiles', done => {
...
         tr._calcMatrices();
         const sourceCache = createSourceCache({noLoad: true});
         sourceCache.transform = tr;
-        sourceCache.onAdd();
+        sourceCache.onAdd(undefined);
         expect(sourceCache.tilesIn([
             new Point(0, 0),
             new Point(512, 256)
-        ], 10, tr)).toEqual([]);
+        ], 10, tr as any as boolean)).toEqual([]);
```
Comment (HarelM) [inline src/source/source_cache.test.ts:1228]: This is strange... Casting to Boolean...?

File: src/source/source_cache.test.ts [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
 import {extend} from '../util/util';
 import browser from '../util/browser';
+import Dispatcher from '../util/dispatcher';
 
 // Add a mocked source type for use in these tests
 function MockSourceType(id, sourceOptions, _dispatcher, eventedParent) {
     // allow tests to override mocked methods/properties by providing
     // them in the source definition object that's given to Source.create()
     class SourceMock extends Evented {
+        id: any;
```
Comment (HarelM) [inline src/source/source_cache.test.ts:18]: Move this class out of the function and only initit in the function. Also add typings to the function parameter.

**Review by HarelM — CHANGES_REQUESTED** [review]

File: src/source/source_cache.test.ts (hunk 2 above)
Comment (wipfli) [reply src/source/source_cache.test.ts:1228]: Right. Really strange I agree. Since `tr` is an object, I guess this will always cast to `true`.

**Review by HarelM — APPROVED** [review]

**Pull request merged**